### PR TITLE
Refactor cpp, add java implementation for Array Division (cses-1085)

### PR DESCRIPTION
**Pull request by ChuangSheep**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Review by SansPapyrus683 — APPROVED** [review]
冰淇凌

File: solutions/silver/cses-1085.mdx [hunk 1]
```diff
@@ -24,75 +24,128 @@ subarray does not exceed $x$, and check if the number of subarrays is $\leq k$.
...
+/**
+ * true if the given `arr` can be divided into `k` subarrays where the sum of
+ * each subarray is at most `max_sum`
+ */
+bool is_possible(const vector<ll> &arr, const int k, ll max_sum) {
+	// # of subarrays needed if sum of each subarray is at most max_sum
+	int subarr_count = 0;
+	// sum of the current subarray
+	ll cur_sum = 0;
+	for (const int &x : arr) {
```
Comment (ryanchou-dev) [inline solutions/silver/cses-1085.mdx:55]:
```suggestion
	ll cur_sum = 0;

	for (const int &x : arr) {
```

File: solutions/silver/cses-1085.mdx [hunk 2]
```diff
@@ -2,7 +2,7 @@
 id: cses-1085
 source: CSES
 title: Array Division
-author: Michael Cao, George Pong
+author: Michael Cao, George Pong, Chuyang Wang
 ---
 
 # Explanation
```
Comment (ryanchou-dev) [inline solutions/silver/cses-1085.mdx:8]:
```suggestion
## Explanation
```

**Review by ryanchou-dev — APPROVED** [review]
dubs on dubs

**Pull request merged**